### ISSUE TITLE
Display the complete list of blog posts

### ISSUE THREAD

**Issue by julien-deramond** [HiDeoo/starlight-blog#104]
### Is your feature request related to a problem?

When navigating through blog posts—whether as a reader or the author—the _All Posts_ link is a helpful way to access recent posts and explore them individually. However, I often find myself wanting a complete list of all posts to quickly get an overview of what's available without relying on paginated navigation.

### Describe the solution you'd like

I'm not entirely sure if this is the best solution, but having a dedicated page that lists all blog posts in one place would be incredibly useful. This feature could be optional, as it might not be necessary for all users. Within the sidebar structure, it could function as a sub-entry under the _All Posts_ section.

**Comment by HiDeoo** [comment]
Thanks for the feedback, always appreciated :raised_hands:

I think there are 2 answers to the use case:

> However, I often find myself wanting a complete list of all posts to quickly get an overview of what's available without relying on paginated navigation.

This feels to me more like an author feature, and I think this could be workaround right now using the [`postCount`](https://starlight-blog-docs.vercel.app/configuration#postcount) and [`recentPostCount`](https://starlight-blog-docs.vercel.app/configuration#recentpostcount) options:

```js
starlightBlog({
  // Use Infinity to show all posts during development
  recentPostCount: import.meta.env.DEV ? Infinity : 10,
  postCount: import.meta.env.DEV ? Infinity : 5,
});
```

_Note that passing `Infinity` requires at least the version [`0.16.1`](https://github.com/HiDeoo/starlight-blog/releases/tag/starlight-blog%400.16.1) of the plugin._

On a more general note, I think this would be a duplicate of https://github.com/HiDeoo/starlight-blog/issues/90. The idea in the future would be to expose all blog related data, e.g. `Astro.locals.starlightBlog.posts` or `Astro.locals.starlightBlog.tags`, etc. and then anyone could build a custom page with all the posts, tags, etc. they want.

As mentioned in the linked issue, this is something that will only happen after Starlight own route data is moved to Astro's `locals` object, which is a work in progress.

Hope this gives a "good enough" workaround for now, and a glimpse of what's to come in the future. Let me know if you have any other questions or feedback.

**Comment by julien-deramond** [comment]
Thanks for your feedback and analysis HiDeoo 🙌

> On a more general note, I think this would be a duplicate of https://github.com/HiDeoo/starlight-blog/issues/90.

That's a good point. I hesitated because the original issue seemed to focus more on the technical aspect, while I wanted to suggest a slightly different rendering option as well. However, both ideas are definitely aligned in spirit.

> This feels to me more like an author feature[...]

Possibly! For now, I've set the "All posts" section to `Infinity`, as I think it's still useful for users given the relatively small number of posts. In the future, I'm considering adding a dedicated page that lists only the blog post titles. Alternatively, I might implement a rendering option within the same "All posts" page to display just the titles.
I'd say that allowing users to quickly browse through all posts or navigate in batches of 50, depending on their preference, can be useful to discover the content and the topics of a blog.
